### PR TITLE
Kubevirt etcd-storage zvol encrypted with vault

### PR DESCRIPTION
**Pull request by andrewd-zededa**
Also cleanup and organize constants.

Previous implementation left etcd-storage open, this PR causes new installs to encrypt etcd-storage identically to vault.

### CODE REVIEW

File: pkg/pillar/zfs/zfs.go [hunk 1]
```diff
@@ -127,11 +121,11 @@ func CreateVaultVolumeDataset(log *base.LogObject, datasetName string, zfsKeyFil
 	props[libzfs.DatasetPropVolsize] = libzfs.Property{
 		Value: strconv.FormatUint(alignedSize, 10)}
 	props[libzfs.DatasetPropVolblocksize] = libzfs.Property{
-		Value: strconv.FormatUint(VolBlockSize, 10)}
+		Value: strconv.FormatUint(blockSize, 10)}
 	props[libzfs.DatasetPropVolmode] = libzfs.Property{
 		Value: "dev"}
 	props[libzfs.DatasetPropCompression] = libzfs.Property{
-		Value: "zstd"}
+		Value: compression}
```
Comment (zedi-pramodh) [inline pkg/pillar/zfs/zfs.go:128]: What is this change about ?  I thought we use zstd compression, why remove it ?

Comment (zedi-pramodh) [reply pkg/pillar/zfs/zfs.go:128]: OK I understand the change, you are explicitly saying compression "off" but why are we turning off compression for this zvol ? Performance issues ?

Comment (zedi-pramodh) [reply pkg/pillar/zfs/zfs.go:128]: Ok can you rename that to compression_type ? In that way its clear the type we are using zstd or off. Just compression implies compression is enabled.

Comment (andrewd-zededa) [reply pkg/pillar/zfs/zfs.go:128]: how about more camel-case like compressionType to match surrounding styling?

**Review by zedi-pramodh — APPROVED** [review]
LGTM

**Pull request merged**